"I'm done"

martyav opened the pull request:
Not really, but at this moment I am moving around deckchairs on the Titanic

martyav committed:
first commit

martyav committed:
working out cell configuration

martyav committed:
Attempts were made at creating functions that would parse data and make API calls; also added an imageView to the detail view

martyav committed:
working on function that makes object Array of BricABracs

martyav committed:
fixing some slight typos from immediately previous commit

martyav committed:
added an API Helper function based on code from https://github.com/jerjunkel/AC3.2-NSURLSession/blob/master/InstaDogFactory.swift

martyav committed:
trying to fix problems with apihelper call

martyav committed:
git commit -a -m "I'm done"

Not really done, but at this moment I am moving deckchairs on the Titanic.

jgresh commented on VictoriaAlbert/APIHelper.swift:
```diff
@@ -0,0 +1,33 @@
+
+//this is modified from my instaCats project: https://github.com/jerjunkel/AC3.2-NSURLSession/blob/master/InstaDogFactory.swift
+
+func APIHelper(endpoint: String, callback: @escaping ([BricABrac]?) -> ()) {
```
Let's move over to the class-based APIManager. It works much in the same way as this.

jgresh commented on VictoriaAlbert/TableViewController.swift:
```diff
@@ -0,0 +1,120 @@
+class TableViewController: UITableViewController {
+    
+    var selectedBric: BricABrac?
+    var bricArr:[BricABrac]? = [BricABrac]()
```
This line has a couple competing ideas. You're declaring `bricArr` to be an Optional array of `BricABrac` but then you initialize it to an emty array of `BricABrac`. That's not the same as nil. So, unless you plan to nil it out later (which I don't think you do or have to) you can remove the optional aspect of it. Then it's a little easier to deal with and look at later in the code.

jgresh commented on VictoriaAlbert/VictoriaAlbert/BricABrac.swift:
```diff
@@ -0,0 +1,112 @@
+            }
+            print("We made \(theBigBox)")
+            
+            guard let records: [String: Any?] = castTheBox["records"] as? [String : Any?] else {
```
This was a `[[String:Any]]` not `[String:Any?]`.

jgresh commented on VictoriaAlbert/VictoriaAlbert/BricABrac.swift:
```diff
@@ -0,0 +1,112 @@
+            
+            for item in records {
+                // subtitles are easy
+                //oh i need to get into the fields...which are an array of dictionaries...
```
Yes, this comment is correct. Did you run out of time/energy?

jgresh commented on VictoriaAlbert/VictoriaAlbert/Info.plist:
```diff
@@ -0,0 +1,43 @@
+	</array>
+	<key>NSAppTransportSecurity</key>
+	<dict>
+		<key>NSAllowsArbitraryLoadsInWebContent</key>
```
Unfortunately, there is a small error in this key that prevented you from getting data which made your code untestable, which must have been maddening for you. `NSAllowsArbitraryLoadsInWebContent` should have been `NSAllowsArbitraryLoads`.

jgresh commented on VictoriaAlbert/VictoriaAlbert/Pic.swift:
```diff
@@ -0,0 +1,40 @@
+    init(_ imageID:String) {
+        self.imageID = imageID
+        var secondPart = ""
+        for char in imageID.characters {
```
While this didn't get you all the way there, I like your approach here.

martyav committed:
fixed model and plist on jason's advice re: networking

+ changed key in plist: NSAllowsArbitraryLoadsInWebContent ---> NSAllowsArbitraryLoads.
+ altered guard-lets so we can drill into the fields and pull the info we want

martyav committed:
at this point, we have the function makeBricABrac working, but we are having issues loading the data

martyav committed:
trying to fix the casting

martyav committed:
casting from titles is breaking us
